### PR TITLE
Moving Ace Init

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Moves the ACE initialization code to it's own function. Adds some more logging of specific actions that players can do with medical items.

### Please specify which Issue this PR Resolves.
N/A

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### CODE REVIEW

**Comment by killerswin2** [comment]
Finally, had to fix my fork for it to work right!

File: A3A/addons/core/functions/init/fn_initClient.sqf [hunk 1]
```diff
@@ -71,7 +71,6 @@ if (!isServer) then {
     [clientOwner, "destroyedBuildings"] remoteExecCall ["publicVariableClient", 2];
 
     boxX call jn_fnc_arsenal_init;
-    if (A3A_hasACEMedical) then { call A3A_fnc_initACEUnconsciousHandler };
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_initClient.sqf:74]: Problem here is that you've lost the installation of this handler on headless clients.

This handler is different from the others you moved to initACE: The others are all player-specific and don't need installation on a DS or HC, while the unconscious handler needs to be installed everywhere.

Maybe best to rename initACE to something more obviously player-specific, remove the iniACE call from initServer and keep both the initACEUnconscious handler calls where they are.

File: A3A/addons/core/functions/init/fn_initACE.sqf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            // wait in case our event was called first
+            waitUntil {sleep 1; !isNull getAssignedCuratorLogic _unit};
+
+            //now add the logging to the module
+            [[getAssignedCuratorLogic _unit]] remoteExecCall ["A3A_fnc_initZeusLogging",0];
+        };
+    }] call CBA_fnc_addEventHandler;
+};
+
+["ace_explosives_place", {
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_initACE.sqf:48]: While we're adding long-missing event handlers, let's get the ACE throw EH in here to fix the old undercover grenade throw exploit. For some reason I added it to punishment and not initClient.

Something like this:
```
    ["ace_throwableThrown", {
        params ["_unit", "_throwable"];
        if (isCaptive player && _unit == player) then { player setCaptive false };
    }] call CBA_fnc_addEventHandler;
    ```

File: A3A/addons/core/functions/init/fn_initACE.sqf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            params ["_unit"];
+
+            // wait in case our event was called first
+            waitUntil {sleep 1; !isNull getAssignedCuratorLogic _unit};
+
+            //now add the logging to the module
+            [[getAssignedCuratorLogic _unit]] remoteExecCall ["A3A_fnc_initZeusLogging",0];
+        };
+    }] call CBA_fnc_addEventHandler;
+};
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_initACE.sqf:46]: If we're restricting this file to player stuff then move this back to initServer (or another file), I guess.

File: A3A/addons/core/functions/init/fn_initACE.sqf [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ * Example:
+ * call A3A_fnc_initACE;
+ *
+ * Public: No
+ */
+
+#include "..\..\script_component.hpp"
+
+if (A3A_hasACEMedical) then { 
+    call A3A_fnc_initACEUnconsciousHandler;
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_initACE.sqf:22]: And lose this one.

**Review by jaj22 — CHANGES_REQUESTED** [review]

File: A3A/addons/core/functions/init/fn_initACE.sqf [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    // log atropine, epinephrine, and morphine use
+    ["ace_treatmentStarted", {
+        params ["_caller", "_target", "_selectionName", "_className", "_itemUser", "_usedItem"];
+        systemChat str _this;
+        if(_usedItem in ["ACE_atropine", "ACE_epinephrine", "ACE_morphine"]) then {
```
Comment (rautamiekka) [inline A3A/addons/core/functions/init/fn_initACE.sqf:28]:
```suggestion
        if (_usedItem in ["ACE_atropine", "ACE_epinephrine", "ACE_morphine"]) then {
```

File: A3A/addons/core/functions/init/fn_initACE.sqf [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            ServerInfo_3("Player: %1 used %2 on %3",name _caller,_usedItem,name _target);
+        };
+    }] call CBA_fnc_addEventHandler;
+};
+
+if(isServer) then {
```
Comment (rautamiekka) [inline A3A/addons/core/functions/init/fn_initACE.sqf:34]:
```suggestion
if (isServer) then {
```

File: A3A/addons/core/functions/init/fn_initClient.sqf [hunk 7]
```diff
@@ -439,6 +412,9 @@ _layer = ["statisticsX"] call bis_fnc_rscLayer;
 //Load the player's personal save.
 [] spawn A3A_fnc_createDialog_shouldLoadPersonalSave;
 
+// call ACE3 initialization stuff
+if(A3A_hasACE) then {call A3A_fnc_initACE};
```
Comment (rautamiekka) [inline A3A/addons/core/functions/init/fn_initClient.sqf:416]:
```suggestion
if (A3A_hasACE) then {call A3A_fnc_initACE};
```

File: A3A/addons/core/functions/init/fn_initServer.sqf [hunk 8]
```diff
@@ -239,6 +236,8 @@ addMissionEventHandler ["EntityKilled", {
     };
 }];
 
+// call ACE3 initialization stuff
+if(A3A_hasACE) then {call A3A_fnc_initACE};
```
Comment (rautamiekka) [inline A3A/addons/core/functions/init/fn_initServer.sqf:240]:
```suggestion
if (A3A_hasACE) then {call A3A_fnc_initACE};
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]
Consistency fixes.

File: A3A/addons/core/functions/init/fn_initACE.sqf (hunk 4 above)
Comment (jaj22) [reply A3A/addons/core/functions/init/fn_initACE.sqf:22]: Missed this one btw. It doesn't have multiple init protection so you can't call it here.

**Comment by jaj22** [comment]
Fixed a bug in the throwableThrown EH, cleared out the surplus initACEUnconsciousHandler call and did some minor cleanup. Should be good to go now.

Tested on both localhost and DS with ACE:
- Logging with ACE-created zeus.
- Action-blocking on HQ objects and rebel group join/leave.
- Injector usage logging.
- Undercover removal on ACE throw.

 Non-ACE behaviour should be unchanged.

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Pull request merged**